streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -186,67 +150,21 @@ public void register(StateStore store, boolean loggingEnabled, StateRestoreCallb
             }
         } else {
             log.trace("{} Restoring state store {} from changelog topic {}", logPrefix, store.name(), topic);
-
-            restoreActiveState(topic, stateRestoreCallback);
+            final StateRestorer restorer = new StateRestorer(storePartition,
+                                                                    stateRestoreCallback,
+                                                                    checkpointedOffsets.get(storePartition),
+                                                                    offsetLimit(storePartition));
+            // save the restorer for later if it is persistent as we might
+            // need to retrieve the checkpoint offset (in the case where it is never updated during processing).
+            if (store.persistent()) {
```

Comment:
See my comment below, maybe we can just maintain the `restoredOffset` in `Restorer` to get rid of this check?